app/src/main/java/cm/aptoide/pt/network/model/IdsRepository.kt

```diff
@@ -0,0 +1,91 @@
+package cm.aptoide.pt.network.model
+
+import android.content.Context
+import android.os.Looper
+import androidx.annotation.WorkerThread
+import androidx.datastore.core.DataStore
+import androidx.datastore.preferences.core.Preferences
+import androidx.datastore.preferences.core.edit
+import androidx.datastore.preferences.core.stringPreferencesKey
+import com.google.android.gms.ads.identifier.AdvertisingIdClient
+import kotlinx.coroutines.flow.first
+import kotlinx.coroutines.flow.map
+import kotlinx.coroutines.runBlocking
+import timber.log.Timber
+import java.util.UUID
+import javax.inject.Inject
+import javax.inject.Singleton
+
+@Singleton
+class IdsRepository @Inject constructor(
+  private val context: Context,
+  private val dataStore: DataStore<Preferences>,
+) {
+  @Synchronized fun getUniqueIdentifier(): String {
+    val aptoideId: String = getAndroidClientUUID()
+
+    // if we already have the aptoide client uuid, return it
+    if (aptoideId.isNotEmpty()) {
+      Timber.tag(TAG).v("getUniqueIdentifier: in datastore: $aptoideId")
+      return aptoideId
+    }
+
+    val id = getGoogleAdvertisingId()
+    val res = if (!id.isNullOrEmpty())
+      id else UUID.randomUUID().toString()
+    setAndroidClientUUID(res)
+    return res
+  }
+
+  @WorkerThread @Synchronized fun getGoogleAdvertisingId(): String? {
+    val id = getGoogleAdvertisingIDClient()
+    val res = when {
+      id.isNotEmpty() -> id
+      !AdNetworkUtils.isGooglePlayServicesAvailable(context) -> ""
+      Looper.getMainLooper().thread === Thread.currentThread() ->
+        throw IllegalStateException("You cannot run this method from the main thread")
+
+      else -> try {
+        AdvertisingIdClient.getAdvertisingIdInfo(context).id
+      } catch (e: Exception) {
+        throw IllegalStateException(e)
+      }
+    }
+    if (!res.isNullOrEmpty()) {
+      setGoogleAdvertisingIDClient(res)
+    }
+    return res
+  }
+
+  @Synchronized private fun getGoogleAdvertisingIDClient(): String {
+    var aptoideId = ""
+    runBlocking {
+      aptoideId = dataStore.data.map { it[GOOGLE_ADVERTISING_ID_CLIENT] ?: "" }.first()
+    }
+    return aptoideId
+  }
+
+  @Synchronized private fun setGoogleAdvertisingIDClient(value: String): Unit =
+    runBlocking {
```

Comment:
Lets avoid using `runBlocking`

Reply:
I didn't understand if i should totally remove runBlocking or just remove it from IdsRepository, so i just removed from IdsRepository